src/VecSim/spaces/CMakeLists.txt

```diff
@@ -17,74 +16,56 @@ set(OPTIMIZATIONS "")
 if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "(x86_64)|(AMD64|amd64)|(^i.86$)")
 	# Check that the compiler supports instructions flag.
 	# from gcc14+ -mavx512bw is implicitly enabled when -mavx512vbmi2 is requested
-	CHECK_CXX_COMPILER_FLAG(-mavx512vl CXX_AVX512VL)
-	CHECK_CXX_COMPILER_FLAG(-mavx512bf16 CXX_AVX512BF16)
-	CHECK_CXX_COMPILER_FLAG(-mavx512bw CXX_AVX512BW)
-	CHECK_CXX_COMPILER_FLAG(-mavx512vbmi2 CXX_AVX512VBMI2)
-	CHECK_CXX_COMPILER_FLAG(-mavx512f CXX_AVX512F)
-	CHECK_CXX_COMPILER_FLAG(-mavx2 CXX_AVX2)
-	CHECK_CXX_COMPILER_FLAG(-mavx CXX_AVX)
-	CHECK_CXX_COMPILER_FLAG(-mf16c CXX_F16C)
-	CHECK_CXX_COMPILER_FLAG(-mfma CXX_FMA)
-	CHECK_CXX_COMPILER_FLAG(-msse3 CXX_SSE3)
-	CHECK_CXX_COMPILER_FLAG(-msse CXX_SSE)
+	include(${root}/cmake/x86_64InstructionFlags.cmake)
```

Comment:
If you decide not to rename the file, consider commenting that it will define the `CXX_*` conditions we use here and the compilation defiitions